src/autoRetry.coffee

```diff
@@ -0,0 +1,83 @@
+config = require "./config/config"
+config.alerts = config.get('alerts')
+logger = require "winston"
+moment = require 'moment'
+Q = require 'q'
+Channels = require('./model/channels')
+Channel = Channels.Channel
+Transaction = require('./model/transactions').Transaction
+Task = require('./model/tasks').Task
+authorisation = require('./middleware/authorisation')
+utils = require './utils'
+
+
+getChannels = (callback) -> Channel.find autoRetryEnabled: true, status: 'enabled', callback
+
+findTransactions = (channel, callback) ->
+  to = moment().subtract channel.autoRetryPeriodMinutes-1, 'minutes'
+
+  query =
+    $and: [
+        'request.timestamp':
+          $lte: to.toDate()
+      ,
+        channelID: channel._id
+      ,
+        internalServerError: true
+      ,
+        wasRerun: false
+    ]
+
+  logger.debug "Executing query transactions.find(#{JSON.stringify query})"
+  Transaction.find query, _id: 1, callback
+
+createRerunTask = (transactionIDs, callback) ->
+  logger.info "Rerunning failed transactions: #{transactionIDs}"
+  task = new Task
+    transactions: (transactionIDs.map (t) -> tid: t )
+    totalTransactions: transactionIDs.length
+    remainingTransactions: transactionIDs.length
+    user: 'internal'
+
+  task.save (err) ->
+    logger.error err if err
+    callback()
+
+autoRetryTask = (job, done) ->
+  _taskStart = new Date()
+
+  getChannels (err, results) ->
+    promises = []
+
+    for channel in results
+      do (channel) ->
+        deferred = Q.defer()
+
+        findTransactions channel, (err, results) ->
+          if err
+            logger.error err
+            deferred.resolve()
+          else if results? and results.length>0
+            createRerunTask (results.map((r) -> r._id)), -> deferred.resolve()
```

Comment:
Atm it excludes transactions in the search that are too new (`to` date in search) based on the autoRetryPeriod, which is a channel specific setting. I'm sure it's possible to use a single task though; but it will make the logic a bit more complex.